File: frontend/src/services/authService.js
import api from './api';
export const register = (email, password, fullName) =>
  api.post('/api/v1/auth/register', { email, password, fullName });
export const login = (email, password) =>
  api.post('/api/v1/auth/login', { email, password });
export const refresh = (refreshToken) =>
  api.post('/api/v1/auth/refresh', { refreshToken });

File: backend/src/main/java/com/webhookplatform/service/AuthService.java
package com.webhookplatform.service;

import com.webhookplatform.dto.request.LoginRequest;
import com.webhookplatform.dto.request.RegisterRequest;
import com.webhookplatform.dto.response.AuthResponse;

public interface AuthService {
    AuthResponse register(RegisterRequest request);
    AuthResponse login(LoginRequest request);
    AuthResponse refresh(String refreshToken);
}
